public/app/features/trails/MetricSelectScene.tsx

```diff
@@ -181,11 +294,41 @@ function getMetricNamesVariableSet() {
 function getPreviewPanelFor(metric: string, index: number) {
   const autoQuery = getAutoQueriesForMetric(metric);
 
-  return autoQuery.preview
+  const p = autoQuery.preview
     .vizBuilder(autoQuery.preview)
     .setColor({ mode: 'fixed', fixedColor: getColorByIndex(index) })
     .setHeaderActions(new SelectMetricAction({ metric, title: 'Select' }))
     .build();
+
+  const data = p.state.$data;
+  if (data instanceof SceneQueryRunner) {
+    data.addActivationHandler(() => {
+      data.subscribeToState((state) => {
+        if (state._hasFetchedData) {
+          const scene = sceneGraph.getAncestor(p, MetricSelectScene);
+
+          if (!state.data?.series.length) {
+            scene.hideMetric(metric);
+            return;
+          }
+          const valueField = state.data?.series.at(0)?.fields.at(1);
+          const allNull = !valueField?.values.find((v) => v !== null);
+          if (allNull) {
+            scene.hideMetric(metric);
+            return;
+          }
+          const allZero = !valueField?.values.find((v) => v !== 0);
+          if (allZero) {
+            scene.hideMetric(metric);
+            return;
+          }
+          scene.metricHasLoaded(metric);
+        }
+      });
+    });
+  }
```

Comment:
We can do this with a simple behavior on the panel level, we have a POC of this in the scene lib. Will try this out.

Reply:
@adrapereira it's not the graphs that are expensive to render but he queries that are expensive to issue, so showing 30 graphs could then mean 10000 queries it's not ideal :) 

I think we might be better of lazy rendering of the grid items (so only mount and render them when they are visible), that way we can guarantee that we only issue queries for the visible panels. 

right now it's only the [GridLayout](https://github.com/grafana/scenes/blob/main/packages/scenes/src/components/layout/grid/SceneGridLayoutRenderer.tsx#L58) that supports lazy rendering, it's using IntersectionObserver to know when a panel is outside viewport. We need something like it for CSSGridLayout.